src/RageFileManager.cpp

```diff
@@ -227,42 +233,37 @@ bool RageFileManager::Unzip(const std::string &zipPath, std::string targetPath,
 	return success;
 }
 
-/* Wait for the given driver to become unreferenced, and remove it from the list
- * to get exclusive access to it.  Returns false if the driver is no longer available
- * (somebody else got it first). */
-#if 0
-static bool GrabDriver( RageFileDriver *pDriver )
+static void NormalizePath( RString &sPath )
 {
-	g_Mutex->Lock();
-
-	for(;;)
+	FixSlashesInPlace( sPath );
+	CollapsePath( sPath, true );
+	if (sPath.size() == 0)
 	{
-		unsigned i;
-		for( i = 0; i < g_pDrivers.size(); ++i )
-		{
-			if( g_pDrivers[i]->m_pDriver == pDriver )
-			{
-				break;
-			}
-		}
-		if( i == g_pDrivers.size() )
-		{
-			g_Mutex->Unlock();
-			return false;
-		}
+		sPath = '/';
+	}
+	else if (sPath[0] != '/')
+	{
+		sPath = '/' + sPath;
+	}
+}
 
-		if( g_pDrivers[i]->m_iRefs == 0 )
-		{
-			g_pDrivers.erase( g_pDrivers.begin()+i );
-			return true;
-		}
+void RageFileManager::ProtectPath(const std::string& path)
+{
+	RString normalizedPath(path);
+	NormalizePath(normalizedPath);
+	normalizedPath.MakeLower();
 
-		/* The driver is in use.  Wait for somebody to release a driver, and
-		 * try again. */
-		g_Mutex->Wait();
-	}
+	this->protectedPaths.insert(normalizedPath);
```

Comment:
No need to call `this->` (I think it's often just clutter). If we rename the variable to `m_protectedPaths` as suggested below we know it's a member and not a local variable.

Reply:
done